warehouse/index.js

```diff
@@ -92,6 +95,40 @@ function excludeRudderCreatedTableNames(
   return name;
 }
 
+function appendColumnNameAndType(
+  utils,
+  eventType,
+  key,
+  val,
+  output,
+  columnTypes,
+  options,
+  jsonKey = false
+) {
+  const datatype = getDataType(key, val, options, jsonKey);
```

Comment:
shouldn't we always set the data type as JSON if its a jsonKey and stringify the value. This is to make sure that if its the first event and we create a JSON column for that field if its explicity passed in config. We do not want to create a date field etc if they have given it as json path. Makes sense?

Reply:
At the transformer stage, it still doesn't know whether it is first time or later. Setting above conflicts with PRD doc of supporting json to only arrays and map. Above scenario would be ideal if we change our approach of realising json type i.e.string,number,map & arrays. Just store everything as string
Note: 
- Postgres, Snowflake & RS json field can have number, string, maps, arrays
- BQ storing it as string always. Parser JSON functions are safe executable i.e. returns null instead of failure

Reply:
Valid json entries will be marked as json. 
Datetime string is not marked as json even though it is declared as jsonKey

Reply:
To avoid confusion to customers, datestring is considered as string